src/components/button/button.component.js

```diff
@@ -130,13 +136,15 @@ const renderStyledButton = (buttonProps) => {
 };
 
 const Button = (props) => {
+  const theme = useContext(ThemeContext);
```

Comment:
I haven't , I think that maybe it would be worth to do it but not sure we should connect the changes with this PR as we are removing/replacing props only. I think that we should create a ticket or something for that and talk about it on triage/refinement if this is needed

Reply:
if you do it, then there is no need to have context or icon options in the main component